### PR TITLE
feat: add stepper and step components

### PR DESCRIPTION
**Pull request by anusha-c18**
## Done

- Added stepper wrapper component
- Added step component with options to:
   - add a title for the step 
   - enable/disable the step
   - select the step
   - display an optional label
   - display an optional link
   - display a progress line
   - display the index of the step
   - display an icon for the step
- Added tests for the step component
- Added tests for the stepper component 

The current stepper component is of the standard size outlined in the spec. If required, the large size can be added at a later time.

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Steps for QA.

### Percy steps

- List any expected visual change in Percy, or write something like "No visual changes expected" if none is expected.

## Fixes

Fixes: # .

### CODE REVIEW

File: src/components/Stepper/Step/Step.scss [hunk 1]
```diff
@@ -0,0 +1,115 @@
+@import "vanilla-framework";
+
+.p-list__item {
+  padding-bottom: 0;
+  padding-top: 0;
+}
+
+.p-inline-list {
+  display: flex;
+}
```
Comment (edlerd) [inline src/components/Stepper/Step/Step.scss:10]: Those look like standard vanilla classes, which we shouldn't override globally. 

I think best is 
1. to namespace all selectors in this file with a wrapper and 
2. target only custom classes

File: src/components/Stepper/Step/Step.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+   */
+  enabled: boolean;
+  /**
+   * Optional value to highlight the selected step.
+   */
+  selected?: boolean;
+  /**
+   * Icon to display in the step. Specify "number" if the index should be displayed.
+   */
+  iconName: string;
```
Comment (edlerd) [inline src/components/Stepper/Step/Step.tsx:40]: Should we reference the icon names `ValueOf<typeof ICONS> | string | "number"` here to allow autocomplete?

**Review by edlerd — COMMENTED** [review]
QA LGTM.
The linter has some errors, one idea for improvement  and one concern regarding css namespacing.

File: src/components/Stepper/Step/Step.scss (hunk 1 above)
Comment (anusha-c18) [reply src/components/Stepper/Step/Step.scss]: Agreed, I've shifted the css into the `stepper-vertical` and `stepper-horizontal` classes.

File: src/components/Stepper/Step/Step.tsx (hunk 2 above)
Comment (anusha-c18) [reply src/components/Stepper/Step/Step.tsx:40]: While having autocomplete would be nice, adding the general `string` type to the union expands the type, which prevents ts from providing autocomplete suggestions.

File: src/components/Stepper/Step/Step.scss [hunk 3]
```diff
@@ -0,0 +1,113 @@
+@import "vanilla-framework";
+
+.p-inline-list__item {
+  margin: 0;
+}
```
Comment (edlerd) [inline src/components/Stepper/Step/Step.scss:5]: This is overriding a vanilla class on the root level. It should be namespaced.

**Review by edlerd — CHANGES_REQUESTED** [review]

File: src/components/Stepper/Step/Step.scss (hunk 3 above)
Comment (anusha-c18) [reply src/components/Stepper/Step/Step.scss]: Good catch - addressed the [changes](https://github.com/canonical/react-components/compare/ea71adb0101ff639008ef938152a337c3bf734bc..00fb6effdfa8ff1b63b7af74f3ee26d201ce47c7)

**Review by edlerd — APPROVED** [review]
QA and Code LGTM.

We need to have a look at percy and eventually approve the new screenshot introduced by this PR.

**Pull request merged**